src/cmd/prepare.go

```diff
@@ -94,15 +95,43 @@ var prepareFindImages = &cobra.Command{
 	},
 }
 
+var prepareGenerateConfigFile = &cobra.Command{
+	Use:     "generate-config [FILENAME]",
+	Aliases: []string{"gc"},
+	Args:    cobra.MaximumNArgs(1),
+	Short:   "Generates a config file for Zarf",
+	Long: "Generates a Zarf config file for controlling how the Zarf CLI operates. Optionally accepts a filename to write the config to.\n\n" +
+		"The extension will determine the format of the config file, e.g. env-1.yaml, env-2.json, env-3.toml etc. \n" +
+		"Accepted extensions are json, toml, yaml.\n\n" +
+		"NOTE: This file must not already exist. If no filename is provided, the config will be written to the current working directory as zarf-config.toml.",
+	Run: func(cmd *cobra.Command, args []string) {
+		fileName := "zarf-config.toml"
+
+		// If a filename was provided, use that
+		if len(args) > 0 {
+			fileName = args[0]
+		}
+
+		if err := v.SafeWriteConfigAs(fileName); err != nil {
+			message.Fatalf(err, "Unable to write the config file %s, make sure the file doesn't already exist", fileName)
+		}
+	},
+}
+
 func init() {
+	initViper()
+
 	rootCmd.AddCommand(prepareCmd)
 	prepareCmd.AddCommand(prepareTransformGitLinks)
 	prepareCmd.AddCommand(prepareComputeFileSha256sum)
 	prepareCmd.AddCommand(prepareFindImages)
+	prepareCmd.AddCommand(prepareGenerateConfigFile)
+
+	v.SetDefault(V_PKG_CREATE_SET, map[string]string{})
 
 	prepareFindImages.Flags().StringVarP(&repoHelmChartPath, "repo-chart-path", "p", "", `If git repos hold helm charts, often found with gitops tools, specify the chart path, e.g. "/" or "/chart"`)
+	prepareFindImages.Flags().StringToStringVar(&config.CreateOptions.SetVariables, "set", v.GetStringMapString(V_PKG_CREATE_SET), "Specify package variables to set on the command line (KEY=value)")
```

Comment:
It may be good to add hint text to the decription here since it might not be obvious that `zarf prepare find-images` uses the same configuration key as `zarf package create` (same goes for other reuses)

We could also add specific keys for these commands as well, but either way I would argue it should be communicated to the user.

Reply:
added

Reply:
I think it should also be in the output the user sees (i.e.)

```
Specify package variables to set on the command line as done for package create (KEY=value)
```

Reply:
I may have misunderstood, I only added it to the non-obvious one (prepare).  The `generate-config` was designed to give users an example.  Are you saying we should include more details about the config file beyond the docs?

![Screen Shot 2022-10-04 at 7 31 34 PM](https://user-images.githubusercontent.com/882485/193954579-7facb795-b2d1-4da7-8bfc-62a4bf8c258d.png)